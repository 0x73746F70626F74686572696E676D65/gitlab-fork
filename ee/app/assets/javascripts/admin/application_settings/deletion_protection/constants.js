import { __, s__ } from '~/locale';

export const I18N_DELETION_PROTECTION = {
  heading: s__('AdminSettings|Deletion protection'),
  helpText: s__(
    'AdminSettings|Retention period that deleted groups and projects will remain restorable. Personal projects are always deleted immediately. Some groups can opt-out their projects.',
  ),
  learnMore: __('Learn more.'),
  keepDeleted: s__('AdminSettings|Keep deleted'),
  deleteImmediately: s__('AdminSettings|None, delete immediately'),
  for: __('for'),
  days: __('days'),
  groupsOnly: __('groups only'),
  groupsAndProjects: __('groups and projects'),
};
