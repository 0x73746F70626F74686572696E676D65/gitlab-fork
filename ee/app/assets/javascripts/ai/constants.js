import { s__, sprintf } from '~/locale';

export const i18n = {
  GENIE_TOOLTIP: s__('AI|What does the selected code mean?'),
  GENIE_NO_CONTAINER_ERROR: s__("AI|The container element wasn't found, stopping AI Genie."),
  GENIE_CHAT_TITLE: s__('AI|Code Explanation'),
  GENIE_CHAT_CLOSE_LABEL: s__('AI|Close the Code Explanation'),
  GENIE_CHAT_LEGAL_NOTICE: sprintf(
    s__(
      'AI|You are not allowed to copy any part of this output into issues, comments, GitLab source code, commit messages, merge requests or any other user interface in the %{gitlabOrg} or %{gitlabCom} groups.',
    ),
    { gitlabOrg: '<code>/gitlab-org</code>', gitlabCom: '<code>/gitlab-com</code>' },
    false,
  ),
  GENIE_CHAT_LEGAL_GENERATED_BY_AI: s__('AI|Responses generated by AI'),
  REQUEST_ERROR: s__('AI|Something went wrong. Please try again later'),
  EXPERIMENT_BADGE: s__('AI|Experiment'),
  FEEDBACK_LINK: s__('AI|Give feedback on code explanation'),
};

export const AI_GENIE_DEBOUNCE = 300;

export const FEEDBACK_LINK_URL = 'https://gitlab.com/gitlab-org/gitlab/-/issues/407285';
