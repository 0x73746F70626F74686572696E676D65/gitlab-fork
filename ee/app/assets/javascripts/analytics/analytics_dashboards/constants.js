import { s__ } from '~/locale';

export const FEATURE_PRODUCT_ANALYTICS = 'productAnalytics';

export const PRODUCT_ANALYTICS_FEATURE_DASHBOARDS = ['audience', 'behavior'];

export const EVENTS_TYPES = ['pageViews', 'linkClickEvents', 'events'];

export function isRestrictedToEventType(eventType) {
  return EVENTS_TYPES.includes(eventType);
}

export const PANEL_VISUALIZATION_HEIGHT = '600px';

export const PANEL_DISPLAY_TYPES = {
  DATA: 'data',
  VISUALIZATION: 'visualization',
  CODE: 'code',
};

export const PANEL_DISPLAY_TYPE_ITEMS = [
  {
    type: PANEL_DISPLAY_TYPES.DATA,
    icon: 'table',
    title: s__('Analytics|Data'),
  },
  {
    type: PANEL_DISPLAY_TYPES.VISUALIZATION,
    icon: 'chart',
    title: s__('Analytics|Visualization'),
  },
  {
    type: PANEL_DISPLAY_TYPES.CODE,
    icon: 'code',
    title: s__('Analytics|Code'),
  },
];

export const MEASURE_COLOR = '#00b140';
export const DIMENSION_COLOR = '#c3e6cd';

export const EVENTS_TABLE_NAME = 'TrackedEvents';
export const SESSIONS_TABLE_NAME = 'Sessions';
export const RETURNING_USERS_TABLE_NAME = 'ReturningUsers';

export const TRACKED_EVENTS_KEY = 'trackedevents';

export const ANALYTICS_FIELD_CATEGORIES = [
  {
    name: s__('Analytics|Pages'),
    category: 'pages',
  },
  {
    name: s__('Analytics|Users'),
    category: 'users',
  },
  {
    name: s__('Analytics|Link clicks'),
    category: 'linkClicks',
  },
  {
    name: s__('Analytics|Custom events'),
    category: 'custom',
  },
];

export const ANALYTICS_FIELDS = [
  {
    name: s__('Analytics|URL'),
    category: 'pages',
    dbField: 'pageUrl',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Path'),
    category: 'pages',
    dbField: 'pageUrlpath',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Title'),
    category: 'pages',
    dbField: 'pageTitle',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Language'),
    category: 'pages',
    dbField: 'documentLanguage',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Host'),
    category: 'pages',
    dbField: 'pageUrlhosts',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Referer'),
    category: 'users',
    dbField: 'pageReferrer',
    icon: 'user',
  },
  {
    name: s__('Analytics|Language'),
    category: 'users',
    dbField: 'browserLanguage',
    icon: 'user',
  },
  {
    name: s__('Analytics|Viewport'),
    category: 'users',
    dbField: 'viewportSize',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser Family'),
    category: 'users',
    dbField: 'agentName',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser'),
    category: 'users',
    dbField: ['agentName', 'agentVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|OS'),
    category: 'users',
    dbField: 'osName',
    icon: 'user',
  },
  {
    name: s__('Analytics|OS Version'),
    category: 'users',
    dbField: ['osName', 'osVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|User Id'),
    category: 'users',
    dbField: 'userId',
    icon: 'user',
  },
  {
    name: s__('Analytics|Target URL'),
    category: 'linkClicks',
    dbField: ['targetUrl'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Element ID'),
    category: 'linkClicks',
    dbField: ['elementId'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Event Name'),
    category: 'custom',
    dbField: 'customEventName',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Event Props'),
    category: 'custom',
    dbField: 'customEventProps',
    icon: 'documents',
  },
  {
    name: s__('Analytics|User Props'),
    category: 'custom',
    dbField: 'customUserProps',
    icon: 'user',
  },
];

export const NEW_DASHBOARD = () => ({
  title: '',
  description: '',
  panels: [],
  userDefined: true,
});

export const FILE_ALREADY_EXISTS_SERVER_RESPONSE = 'A file with this name already exists';

export const EVENT_LABEL_CREATED_DASHBOARD = 'user_created_custom_dashboard';
export const EVENT_LABEL_EDITED_DASHBOARD = 'user_edited_custom_dashboard';
export const EVENT_LABEL_VIEWED_DASHBOARD_DESIGNER = 'user_viewed_dashboard_designer';
export const EVENT_LABEL_VIEWED_CUSTOM_DASHBOARD = 'user_viewed_custom_dashboard';

export const EVENT_LABEL_USER_VIEWED_VISUALIZATION_DESIGNER = 'user_viewed_visualization_designer';
export const EVENT_LABEL_USER_CREATED_CUSTOM_VISUALIZATION = 'user_created_custom_visualization';

export const EVENT_LABEL_EXCLUDE_ANONYMISED_USERS = 'exclude_anonymised_users';
