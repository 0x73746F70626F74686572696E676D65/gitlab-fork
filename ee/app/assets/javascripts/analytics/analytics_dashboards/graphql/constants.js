export const TYPENAME_PRODUCT_ANALYTICS_DASHBOARD = 'CustomizableDashboards';
export const TYPENAME_PRODUCT_ANALYTICS_DASHBOARD_CONNECTION = 'CustomizableDashboardsConnection';
