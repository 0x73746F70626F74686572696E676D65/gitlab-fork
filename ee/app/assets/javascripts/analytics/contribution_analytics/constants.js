import { __, s__ } from '~/locale';

export const CHART_HEIGHT = 350;
export const INNER_CHART_HEIGHT = 200;
export const CHART_X_AXIS_ROTATE = 45;
export const CHART_X_AXIS_NAME_TOP_PADDING = 55;

export const LEGACY_TABLE_COLUMNS = [
  { name: 'fullname', text: __('Name') },
  { name: 'push', text: __('Pushed') },
  { name: 'issuesCreated', text: __('Opened issues') },
  { name: 'issuesClosed', text: __('Closed issues') },
  { name: 'mergeRequestsCreated', text: __('Opened MRs') },
  { name: 'mergeRequestsApproved', text: __('Approved MRs') },
  { name: 'mergeRequestsMerged', text: __('Merged MRs') },
  { name: 'mergeRequestsClosed', text: __('Closed MRs') },
  { name: 'totalEvents', text: __('Total Contributions') },
];

export const TABLE_COLUMNS = [
  { key: 'user', label: s__('ContributionAnalytics|Name') },
  { key: 'repoPushed', label: s__('ContributionAnalytics|Pushed') },
  { key: 'issuesCreated', label: s__('ContributionAnalytics|Opened issues') },
  { key: 'issuesClosed', label: s__('ContributionAnalytics|Closed issues') },
  { key: 'mergeRequestsCreated', label: s__('ContributionAnalytics|Opened MRs') },
  { key: 'mergeRequestsApproved', label: s__('ContributionAnalytics|Approved MRs') },
  { key: 'mergeRequestsMerged', label: s__('ContributionAnalytics|Merged MRs') },
  { key: 'mergeRequestsClosed', label: s__('ContributionAnalytics|Closed MRs') },
  { key: 'totalEvents', label: s__('ContributionAnalytics|Total Contributions') },
].map((col) => ({
  ...col,
  sortable: true,
  thClass: 'gl-vertical-align-middle!',
}));
