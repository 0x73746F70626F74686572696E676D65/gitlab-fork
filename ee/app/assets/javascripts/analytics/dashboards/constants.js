import { s__, __ } from '~/locale';
import { helpPagePath } from '~/helpers/help_page_helper';
import { days, percentHundred } from '~/lib/utils/unit_format';
import { thWidthPercent } from '~/lib/utils/table_utility';
import {
  getStartOfDay,
  dateAtFirstDayOfMonth,
  nSecondsBefore,
  nMonthsBefore,
  monthInWords,
} from '~/lib/utils/datetime_utility';
import {
  DEPLOYMENT_FREQUENCY_METRIC_TYPE,
  LEAD_TIME_FOR_CHANGES,
  TIME_TO_RESTORE_SERVICE,
  CHANGE_FAILURE_RATE,
} from 'ee/api/dora_api';
import {
  LEAD_TIME_METRIC_TYPE,
  CYCLE_TIME_METRIC_TYPE,
  ISSUES_METRIC_TYPE,
  DEPLOYS_METRIC_TYPE,
} from '~/api/analytics_api';

const UNITS = {
  PER_DAY: {
    chartUnits: __('/day'),
    formatValue: (value) => days(value, 1, { unitSeparator: '/' }),
  },
  DAYS: {
    chartUnits: __('days'),
    formatValue: (value) => days(value, 1, { unitSeparator: ' ' }),
  },
  PERCENT: {
    chartUnits: '%',
    formatValue: (value) => percentHundred(value, 2),
  },
};

export const DORA_METRICS = {
  [DEPLOYMENT_FREQUENCY_METRIC_TYPE]: {
    label: s__('DORA4Metrics|Deployment Frequency'),
    ...UNITS.PER_DAY,
  },
  [LEAD_TIME_FOR_CHANGES]: {
    label: s__('DORA4Metrics|Lead Time for Changes'),
    invertTrendColor: true,
    ...UNITS.DAYS,
  },
  [TIME_TO_RESTORE_SERVICE]: {
    label: s__('DORA4Metrics|Time to Restore Service'),
    invertTrendColor: true,
    ...UNITS.DAYS,
  },
  [CHANGE_FAILURE_RATE]: {
    label: s__('DORA4Metrics|Change Failure Rate'),
    invertTrendColor: true,
    ...UNITS.PERCENT,
  },
  [LEAD_TIME_METRIC_TYPE]: {
    label: s__('DORA4Metrics|Lead time'),
    invertTrendColor: true,
    ...UNITS.DAYS,
  },
  [CYCLE_TIME_METRIC_TYPE]: {
    label: s__('DORA4Metrics|Cycle time'),
    invertTrendColor: true,
    ...UNITS.DAYS,
  },
  [ISSUES_METRIC_TYPE]: {
    label: s__('DORA4Metrics|New issues'),
    formatValue: (value) => value,
  },
  [DEPLOYS_METRIC_TYPE]: {
    label: s__('DORA4Metrics|Deploys'),
    formatValue: (value) => value,
  },
};

export const DASHBOARD_TITLE = __('Value Streams Dashboard (Beta)');
export const DASHBOARD_DESCRIPTION_GROUP = s__('DORA4Metrics|Metrics comparison for %{name} group');
export const DASHBOARD_DESCRIPTION_PROJECT = s__(
  'DORA4Metrics|Metrics comparison for %{name} project',
);
export const DASHBOARD_NO_DATA = __('No data available');
export const DASHBOARD_LOADING_FAILURE = __('Failed to load');
export const DASHBOARD_FEEDBACK_INFORMATION = s__(
  'DevopsMetricsDashboard|%{strongStart}Beta feature:%{strongEnd} Leave your thoughts in the %{linkStart}feedback issue%{linkEnd}.',
);
export const DASHBOARD_FEEDBACK_LINK = 'https://gitlab.com/gitlab-org/gitlab/-/issues/381787';

export const CHART_GRADIENT = ['#499767', '#5252B5'];
export const CHART_GRADIENT_INVERTED = [...CHART_GRADIENT].reverse();
export const CHART_LOADING_FAILURE = s__('DORA4Metrics|Failed to load charts');

const NOW = new Date();
const CURRENT_MONTH_START = getStartOfDay(dateAtFirstDayOfMonth(NOW));
const PREVIOUS_MONTH_START = nMonthsBefore(CURRENT_MONTH_START, 1);
const PREVIOUS_MONTH_END = nSecondsBefore(CURRENT_MONTH_START, 1);

export const THIS_MONTH = {
  key: 'thisMonth',
  label: s__('DORA4Metrics|Month to date'),
  start: CURRENT_MONTH_START,
  end: NOW,
  thClass: thWidthPercent(20),
};

export const LAST_MONTH = {
  key: 'lastMonth',
  label: monthInWords(nMonthsBefore(NOW, 1)),
  start: PREVIOUS_MONTH_START,
  end: PREVIOUS_MONTH_END,
  thClass: thWidthPercent(20),
};

export const TWO_MONTHS_AGO = {
  key: 'twoMonthsAgo',
  label: monthInWords(nMonthsBefore(NOW, 2)),
  start: nMonthsBefore(PREVIOUS_MONTH_START, 1),
  end: nSecondsBefore(PREVIOUS_MONTH_START, 1),
  thClass: thWidthPercent(20),
};

export const THREE_MONTHS_AGO = {
  key: 'threeMonthsAgo',
  label: monthInWords(nMonthsBefore(NOW, 3)),
  start: nMonthsBefore(PREVIOUS_MONTH_START, 2),
  end: nSecondsBefore(nMonthsBefore(PREVIOUS_MONTH_START, 1), 1),
};

export const DASHBOARD_TIME_PERIODS = [THIS_MONTH, LAST_MONTH, TWO_MONTHS_AGO, THREE_MONTHS_AGO];

// Generate the chart time periods, starting with the oldest first:
// 5 months ago -> 4 months ago -> etc.
export const CHART_TIME_PERIODS = [5, 4, 3, 2, 1, 0].map((monthsAgo) => ({
  end: monthsAgo === 0 ? NOW : nMonthsBefore(NOW, monthsAgo),
  start: nMonthsBefore(NOW, monthsAgo + 1),
}));

export const DASHBOARD_TABLE_FIELDS = [
  {
    key: 'metric',
    label: __('Metric'),
    thClass: thWidthPercent(25),
  },
  ...DASHBOARD_TIME_PERIODS.slice(0, -1),
  {
    key: 'chart',
    label: s__('DORA4Metrics|Past 6 Months'),
    start: nMonthsBefore(NOW, 6),
    end: NOW,
    thClass: thWidthPercent(15),
    tdClass: 'gl-py-2! gl-pointer-events-none',
  },
];

export const METRIC_TOOLTIPS = {
  [DEPLOYMENT_FREQUENCY_METRIC_TYPE]: {
    description: s__(
      'DORA4Metrics|Average number of deployments to production per day. This metric measures how often value is delivered to end users.',
    ),
    groupLink: '-/analytics/ci_cd?tab=deployment-frequency',
    projectLink: '-/pipelines/charts?chart=deployment-frequency',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'deployment-frequency' }),
  },
  [LEAD_TIME_FOR_CHANGES]: {
    description: s__(
      'DORA4Metrics|The time to successfully deliver a commit into production. This metric reflects the efficiency of CI/CD pipelines.',
    ),
    groupLink: '-/analytics/ci_cd?tab=lead-time',
    projectLink: '-/pipelines/charts?chart=lead-time',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'lead-time-for-changes' }),
  },
  [TIME_TO_RESTORE_SERVICE]: {
    description: s__(
      'DORA4Metrics|The time it takes an organization to recover from a failure in production.',
    ),
    groupLink: '-/analytics/ci_cd?tab=time-to-restore-service',
    projectLink: '-/pipelines/charts?chart=time-to-restore-service',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'time-to-restore-service' }),
  },
  [CHANGE_FAILURE_RATE]: {
    description: s__(
      'DORA4Metrics|Percentage of deployments that cause an incident in production.',
    ),
    groupLink: '-/analytics/ci_cd?tab=change-failure-rate',
    projectLink: '-/pipelines/charts?chart=change-failure-rate',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'change-failure-rate' }),
  },
  [LEAD_TIME_METRIC_TYPE]: {
    description: s__('DORA4Metrics|Median time from issue created to issue closed.'),
    groupLink: '-/analytics/value_stream_analytics',
    projectLink: '-/value_stream_analytics',
    docsLink: helpPagePath('user/analytics/value_stream_analytics', {
      anchor: 'view-the-lead-time-and-cycle-time-for-issues',
    }),
  },
  [CYCLE_TIME_METRIC_TYPE]: {
    description: s__(
      "DORA4Metrics|Median time from the earliest commit of a linked issue's merge request to when that issue is closed.",
    ),
    groupLink: '-/analytics/value_stream_analytics',
    projectLink: '-/value_stream_analytics',
    docsLink: helpPagePath('user/analytics/value_stream_analytics', {
      anchor: 'view-the-lead-time-and-cycle-time-for-issues',
    }),
  },
  [ISSUES_METRIC_TYPE]: {
    description: s__('DORA4Metrics|Number of new issues created.'),
    groupLink: '-/issues_analytics',
    projectLink: '-/analytics/issues_analytics',
    docsLink: helpPagePath('user/analytics/issue_analytics'),
  },
  [DEPLOYS_METRIC_TYPE]: {
    description: s__('DORA4Metrics|Total number of deploys to production.'),
    groupLink: '-/analytics/productivity_analytics',
    projectLink: '-/analytics/merge_request_analytics',
    docsLink: helpPagePath('user/analytics/merge_request_analytics'),
  },
};
