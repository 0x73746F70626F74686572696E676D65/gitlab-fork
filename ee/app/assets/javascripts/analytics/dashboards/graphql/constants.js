export const BUCKETING_INTERVAL_ALL = 'ALL';
