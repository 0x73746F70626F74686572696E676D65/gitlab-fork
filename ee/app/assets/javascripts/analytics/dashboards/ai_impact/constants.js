import { pick } from 'lodash';
import { s__ } from '~/locale';
import {
  FLOW_METRICS,
  DORA_METRICS,
  VULNERABILITY_METRICS,
  AI_METRICS,
} from '~/analytics/shared/constants';

import { UNITS, TABLE_METRICS as VSD_TABLE_METRICS } from '../constants';

export const SUPPORTED_FLOW_METRICS = [FLOW_METRICS.CYCLE_TIME, FLOW_METRICS.LEAD_TIME];

export const SUPPORTED_DORA_METRICS = [
  DORA_METRICS.DEPLOYMENT_FREQUENCY,
  DORA_METRICS.CHANGE_FAILURE_RATE,
];

export const SUPPORTED_VULNERABILITY_METRICS = [VULNERABILITY_METRICS.CRITICAL];

export const SUPPORTED_AI_METRICS = [AI_METRICS.CODE_SUGGESTIONS_USAGE_RATE];

export const TABLE_METRICS = {
  ...pick(VSD_TABLE_METRICS, [
    ...SUPPORTED_FLOW_METRICS,
    ...SUPPORTED_DORA_METRICS,
    ...SUPPORTED_VULNERABILITY_METRICS,
  ]),
  [AI_METRICS.CODE_SUGGESTIONS_USAGE_RATE]: {
    label: s__('DORA4Metrics|Code Suggestions usage'),
    units: UNITS.PERCENT,
  },
};
