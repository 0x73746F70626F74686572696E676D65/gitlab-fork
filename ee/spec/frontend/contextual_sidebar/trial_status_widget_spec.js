import { GlLink, GlButton } from '@gitlab/ui';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { WIDGET } from 'ee/contextual_sidebar/components/constants';
import TrialStatusWidget from 'ee/contextual_sidebar/components/trial_status_widget.vue';
import { mockTracking, unmockTracking } from 'helpers/tracking_helper';
import { __ } from '~/locale';

describe('TrialStatusWidget component', () => {
  let wrapper;
  let trackingSpy;

  const { trackingEvents } = WIDGET;
  const trialDaysUsed = 10;
  const trialDuration = 30;

  const findGlLink = () => wrapper.findComponent(GlLink);
  const findLearnAboutFeaturesBtn = () => wrapper.findByTestId('learn-about-features-btn');

  const createComponent = (providers = {}) => {
    return shallowMountExtended(TrialStatusWidget, {
      provide: {
        trialDaysUsed,
        trialDuration,
        navIconImagePath: 'illustrations/gitlab_logo.svg',
        percentageComplete: 10,
        planName: 'Ultimate',
        trialDiscoverPagePath: 'discover-path',
        ...providers,
      },
      stubs: { GlButton },
    });
  };

  beforeEach(() => {
    trackingSpy = mockTracking(undefined, undefined, jest.spyOn);
  });

  afterEach(() => {
    unmockTracking();
  });

  describe('interpolated strings', () => {
    it('correctly interpolates them all', () => {
      wrapper = createComponent();

      expect(wrapper.text()).not.toMatch(/%{\w+}/);
    });
  });

  describe('without the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent();
    });

    it('matches the snapshot for namespace in active trial', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('matches the snapshot for namespace not in active trial', () => {
      wrapper = createComponent({ percentageComplete: 110 });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders without an id', () => {
      expect(findGlLink().attributes('id')).toBe(undefined);
    });

    describe('tracks when the widget menu is clicked', () => {
      it('tracks with correct information when namespace is in an active trial', async () => {
        const { category, label } = trackingEvents.activeTrialOptions;
        await wrapper.findByTestId('trial-widget-menu').trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label,
        });
      });

      it('tracks with correct information when namespace is not in an active trial', async () => {
        wrapper = createComponent({ percentageComplete: 110 });

        const { category, label } = trackingEvents.trialEndedOptions;
        await wrapper.findByTestId('trial-widget-menu').trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label,
        });
      });
    });

    it('does not render Trial twice if the plan name includes "Trial"', () => {
      wrapper = createComponent({ planName: 'Ultimate Trial' });

      expect(wrapper.text()).toMatchInterpolatedText(
        'Ultimate Trial Day 10/30 Learn about features',
      );
    });

    it('shows the expected day 1 text', () => {
      wrapper = createComponent({ trialDaysUsed: 1 });

      expect(wrapper.text()).toMatchInterpolatedText(
        'Ultimate Trial Day 1/30 Learn about features',
      );
    });

    it('shows the expected last day text', () => {
      wrapper = createComponent({ trialDaysUsed: 30 });

      expect(wrapper.text()).toMatchInterpolatedText(
        'Ultimate Trial Day 30/30 Learn about features',
      );
    });
  });

  describe('with the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent({ containerId: 'some-id' });
    });

    it('renders with the given id', () => {
      expect(findGlLink().attributes('id')).toBe('some-id');
    });
  });

  describe('with link to trial discover page', () => {
    it('renders the link', () => {
      wrapper = createComponent();

      expect(wrapper.text()).toContain(__('Learn about features'));
      expect(findLearnAboutFeaturesBtn().exists()).toBe(true);
      expect(findLearnAboutFeaturesBtn().attributes('href')).toBe('discover-path');
    });

    describe('when trial is active', () => {
      it('tracks clicking learn about features button', async () => {
        wrapper = createComponent();

        const { category } = trackingEvents.activeTrialOptions;
        await findLearnAboutFeaturesBtn().trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label: 'learn_about_features',
        });
      });
    });

    describe('when trial is expired', () => {
      it('tracks clicking learn about features link', async () => {
        wrapper = createComponent({ percentageComplete: 110 });

        const { category } = trackingEvents.trialEndedOptions;
        await findLearnAboutFeaturesBtn().trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label: 'learn_about_features',
        });
      });
    });
  });
});
