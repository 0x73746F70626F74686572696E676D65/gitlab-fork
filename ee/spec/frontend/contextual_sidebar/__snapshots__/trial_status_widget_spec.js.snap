// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialStatusWidget component without the optional containerId prop matches the snapshot for namespace in active trial 1`] = `
<gl-link-stub
  href="discover-path"
  title="Ultimate Trial"
>
  <div
    class="gl-align-items-stretch gl-display-flex gl-flex-direction-column gl-w-full"
    data-testid="trial-widget-menu"
  >
    <div>
      <div
        class="gl-align-items-center gl-display-flex gl-w-full"
      >
        <span
          class="gl-mb-1 gl-mr-3 nav-icon-container svg-container"
        >
          <img
            alt=""
            class="svg"
            src="illustrations/gitlab_logo.svg"
            width="16"
          />
        </span>
        <span
          class="gl-flex-grow-1 nav-item-name"
        >
          Ultimate Trial
        </span>
        <span
          class="gl-font-sm gl-mr-auto gl-overflow-hidden gl-whitespace-nowrap"
        >
          Day 10/30
        </span>
      </div>
      <div
        class="gl-align-items-stretch gl-display-flex gl-mt-2"
      >
        <gl-progress-bar-stub
          aria-hidden="true"
          class="gl-flex-grow-1"
          value="10"
        />
      </div>
      <b-button-stub
        class="gl-button gl-mt-3 gl-text-decoration-underline"
        data-testid="learn-about-features-btn"
        href="discover-path"
        size="sm"
        tag="button"
        title="Learn about features"
        type="button"
        variant="link"
      >
        <span
          class="gl-button-text"
        >
          Learn about features
        </span>
      </b-button-stub>
    </div>
  </div>
</gl-link-stub>
`;

exports[`TrialStatusWidget component without the optional containerId prop matches the snapshot for namespace not in active trial 1`] = `
<gl-link-stub
  href="discover-path"
  title="Your 30-day trial has ended"
>
  <div
    class="gl-align-items-stretch gl-display-flex gl-flex-direction-column gl-w-full"
    data-testid="trial-widget-menu"
  >
    <div
      class="gl-display-flex gl-gap-4 gl-px-2 gl-w-full"
    >
      <gl-icon-stub
        class="gl-shrink-0 gl-text-blue-600!"
        name="information-o"
        size="16"
      />
      <div>
        <div
          class="gl-font-weight-bold"
        >
          Your 30-day trial has ended
        </div>
        <div
          class="gl-mt-3"
        >
          Looking to do more with GitLab?
          <b-button-stub
            class="gl-button gl-mb-1 gl-text-black-normal! gl-text-decoration-underline"
            data-testid="learn-about-features-btn"
            href="discover-path"
            size="sm"
            tag="button"
            title="Learn about features"
            type="button"
            variant="link"
          >
            <span
              class="gl-button-text"
            >
              Learn about features
            </span>
          </b-button-stub>
        </div>
      </div>
    </div>
  </div>
</gl-link-stub>
`;
